enumerable.rb

```diff
@@ -0,0 +1,114 @@
+# rubocop:disable Metrics/PerceivedComplexity,Metrics/CyclomaticComplexity
+module Enumerable
+  def my_each(_array = nil)
+    return to_enum unless block_given?
+
+    Array(self).length.times do |i|
+      yield(Array(self)[i])
+    end
+    self
+  end
+
+  def my_each_with_index(_array = nil)
+    return to_enum unless block_given?
+
+    Array(self).length.times do |index|
+      yield(to_a[index], index)
+    end
+    self
+  end
+
+  def my_select
+    return to_enum unless block_given?
+
+    result = []
+    my_each do |item|
+      result.push(item) if yield(item)
+    end
+    result
+  end
+
+  def my_all?(obj = nil)
+    return true if obj.nil? && !block_given?
+
+    result = true
+    if obj.nil?
+      my_each { |i| return !result unless yield(i) }
+    elsif !obj.nil?
+      my_each { |i| return !result unless i.match?(obj) }
+    end
+    result
+  end
+
+  def my_any?(obj = nil)
+    return true if obj.nil? && !block_given?
+
+    result = false
+    if obj.nil?
+      my_each { |i| break result = true if yield(i) }
+    elsif !obj.nil?
+      my_each { |i| break result = true if i.match?(obj) }
+    end
+    result
+  end
+
+  def my_none?(obj = nil)
+    return true if obj.nil? && !block_given?
+
+    result = true
+    if obj.nil?
```

Comment:
- [ ] my_none, when no block or argument is passed, should return false if one of the array elements is true.

- [ ] Also if a class is passed should return the right result.

- [ ] Yours throws an error of `wrong argument type Integer (expected Regexp)` in a case like this `array_of_words.my_none?(5)`.